src/components/Chat/Search.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import { collection, query, where, getDocs } from 'firebase/firestore'
+import { db } from '../../firebase'
+
+const Search = () => {
+  const [user, setUser] = useState('')
+  const [username, setUsername] = useState('')
+
+  const searchUser = async () => {
+    const q = query(collection(db, 'users'), where('username', '==', username))
+
+    try {
+      const querySnapshot = await getDocs(q)
+      querySnapshot.forEach((doc) => {
+        console.log(doc.data())
+        setUser(doc.data())
+      })
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  const handleKey = (e) => {
```

Comment:
Running the search on each keypress is not efficient. Debounce this.
You can read up on it here:
https://blog.logrocket.com/how-and-when-to-debounce-or-throttle-in-react/
Also, heres an example with hooks (use this):
https://usehooks-ts.com/react-hook/use-debounce